src/main/java/com/codegym/jrugotom5/dto/AdvertCreateDTO.java

```diff
@@ -0,0 +1,30 @@
+package com.codegym.jrugotom5.dto;
+
+import com.codegym.jrugotom5.entity.User;
+import lombok.Getter;
+import lombok.Setter;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+import java.time.LocalDate;
+
+@Getter
+@Setter
+public class AdvertCreateDTO {
+
+    @NotNull(message = "Title is required")
+    @Size(min = 5, max = 50, message = "Title must be between 5 and 50 characters")
+    private String title;
+
+    @Size(min = 10, max = 200, message = "Description must be between 10 and 200 characters")
+    private String description;
+
+    @NotNull
+    private User createdBy;
```

Comment:
entity object does not belong in dto. you receive data through enpoint, then it serializes to your dto, entity objects are not serializable by default and they are not meant for such use. 
user id should be there instead. take this id -> userRepository.findById() -> insert user obj into advert obj -> save advert obj.
@Volodymyr95 am i right?

Reply:
I'll wait for mentor to check this issue before fixing

Reply:
> I'll wait for mentor to check this issue before fixing

did you try to call your endpoint with postman?

Reply:
no but as far as im concerned the field createdBy can be pasted like this
    "createdBy": {
        "id": 1
    }
however i read about this issue and yeah, you are right, i will fix the issue ASAP, thanks)